tests/test_integration_http.py

```diff
@@ -606,3 +608,64 @@ def test_capture_minidump(cmake, httpserver):
     assert_attachment(envelope)
 
     assert_minidump(envelope)
+
+
+@pytest.mark.parametrize(
+    "run_args",
+    [
+        pytest.param(["http-proxy"]),  # HTTP proxy test runs on all platforms
+        pytest.param(
+            ["socks5-proxy"],
+            marks=pytest.mark.skipif(
+                sys.platform not in ["darwin", "linux"],
+                reason="SOCKS5 proxy tests are only supported on macOS and Linux",
+            ),
+        ),
+    ],
+)
+@pytest.mark.parametrize("proxy_status", [(["off"]), (["on"])])
+def test_capture_proxy(cmake, httpserver, run_args, proxy_status):
+    if not shutil.which("mitmdump"):
+        pytest.skip("mitmdump is not installed")
+
+    proxy_process = None  # store the proxy process to terminate it later
+
+    try:
+        if proxy_status == ["on"]:
+            # start mitmdump from terminal
+            if run_args == ["http-proxy"]:
+                proxy_process = subprocess.Popen(["mitmdump"])
+                time.sleep(5)  # Give mitmdump some time to start
+                if not is_proxy_running("localhost", 8080):
+                    pytest.fail("mitmdump (HTTP) did not start correctly")
+            elif run_args == ["socks5-proxy"]:
+                proxy_process = subprocess.Popen(["mitmdump", "--mode", "socks5"])
+                time.sleep(5)  # Give mitmdump some time to start
+                if not is_proxy_running("localhost", 1080):
+                    pytest.fail("mitmdump (SOCKS5) did not start correctly")
+
+        tmp_path = cmake(["sentry_example"], {"SENTRY_BACKEND": "none"})
+
+        # make sure we are isolated from previous runs
+        shutil.rmtree(tmp_path / ".sentry-native", ignore_errors=True)
+
+        httpserver.expect_request("/api/123456/envelope/").respond_with_data("OK")
+
+        run(
+            tmp_path,
+            "sentry_example",
+            ["log", "start-session", "capture-event"]
+            + run_args,  # only passes if given proxy is running
+            check=True,
+            env=dict(os.environ, SENTRY_DSN=make_dsn(httpserver)),
+        )
+        if proxy_status == ["on"]:
+            assert len(httpserver.log) == 2
+        elif proxy_status == ["off"]:
+            # Window's Autoproxy will send the request even if the http proxy fails
+            # https://learn.microsoft.com/en-us/windows/win32/winhttp/winhttp-autoproxy-support
```

Comment:
Curiously, doesn't this also affect crashpad transport? Are you sure that the PAC auto proxy configuration is the cause here?

Reply:
I'm not sure, I was also puzzled by the fact that it doesn't affect the crashpad test. This might need some further investigation before merging, as I cannot say for certain that the autoproxy setting is the cause.

Reply:
Upon closer investigation, this [code snippet](https://github.com/getsentry/sentry-native/pull/1063/files#diff-8492f5ddad4505921f8eb4d12bdeba3e62a966aafc94fd235dd77e4aba7c0e80R86-R105) (also pasted below) seemed to imply that there is no fallback on the Windows crashpad proxy crash test (whilst there is one on the proxy-capture test); However, changing the assertion at the end to ensure we only expect a log IF the proxy was running (instead of relying on the `Assertionerror` to handle that situation), the test fails on Windows **and** on macOS because it falls back.

The argument for Apple's fallback stands with the blogpost already mentioned in the `Assertionerror` scope, but there is probably a similar method (potentially the Autoproxy) that results in the same behaviour on Windows. This can use some further investigation from my end to make sure we have the root cause documented.

I'll also adapt the test to more clearly indicate the expected return when the proxy is turned off, given that we now know about these fallback methods.

```diff
        try:
            with httpserver.wait(timeout=10) as waiting:
                child = run(
                    tmp_path, "sentry_example", ["log", "crash"] + run_args, env=env
                )
                assert child.returncode  # well, it's a crash after all
        except AssertionError:
            # only want to end up here on (http-proxy, off) tuple (we expect the request to fail)
            if run_args == ["http-proxy"] and proxy_status == ["off"]:
                assert len(httpserver.log) == 0
                return
            elif run_args == ["socks5-proxy"] and proxy_status == ["off"]:
                # Apple's NSURLSession will send the request even if the socks proxy fails
                # https://forums.developer.apple.com/forums/thread/705504?answerId=712418022#712418022
                assert len(httpserver.log) == (1 if (sys.platform == "darwin") else 0)
                return

        assert waiting.result

+       assert len(httpserver.log) == (1 if proxy_status == ["on"] else 0);
```

Reply:
I've added some comments to the code explaining the observed proxy behaviour: https://github.com/getsentry/sentry-native/pull/1063/commits/22982f83fb56c62de1043fccdc16d52ece2f3d4d

- We see that on the [`capture_proxy` test](https://github.com/getsentry/sentry-native/pull/1063/files#diff-5254164492b3599852a520cb29d9b5cdc964768ea2dfc6ddbd9714a45ee03f6eR663-R668), only Windows provides a fallback (we still expect `0` when `proxy_status=off` for non-win32 platforms). 
- In the [`crash_proxy` test](https://github.com/getsentry/sentry-native/pull/1063/files#diff-8492f5ddad4505921f8eb4d12bdeba3e62a966aafc94fd235dd77e4aba7c0e80R105-R107) on the other hand, regardless of the platform, if the `httpserver.wait` returns we always get an `httpserver.log` of size `1` even when the `proxy_status=off`.

Reply:
So, to be clear, the outcome is the following:

* on `macOS`, we have a different bypass behavior (bypasses in `crashpad_handler` if the proxy is unreachable but honors proxy without in our transport), which is expected because we use the `NS*` networking API in the `crashpad_handler` but defer to `libcurl` in our transport.
* on `Linux`, the proxy is honored without bypass in both, which is expected because both use `libcurl`
* on `Windows`, the proxy is by-passed in both, if the proxy is unreachable because both use WinHTTP

Am I summarizing this correctly? If yes, that should be documented in the sentry header's inline docs. Of course, only the observable behavior should be documented, not my justification.